spec/models/manageiq/providers/azure/cloud_manager/metrics_capture_spec.rb

```diff
@@ -4,58 +4,71 @@
   let(:ems)      { FactoryBot.create(:ems_azure) }
   let(:vm)       { FactoryBot.build(:vm_azure, :ext_management_system => ems, :ems_ref => "my_ems_ref") }
   let(:group)    { FactoryBot.build(:azure_resource_group) }
+  let(:armrest_environment) { double('Armrest environment', :resource_url => 'http://example.com') }
+  let(:armrest_service) { double('::Azure::Armrest::ArmrestService', :environment => armrest_environment) }
 
-  xcontext "#get_counters" do
+  before do
+    allow(ems).to receive(:with_provider_connection).and_yield(armrest_service)
+  end
+
+  context "handles errors" do
     let(:metric) { described_class.new(self) }
-    let(:armrest_service) { double('::Azure::Armrest::ArmrestService') }
 
     before do
       allow(self).to receive(:ext_management_system).and_return(ems)
       allow(self).to receive(:name).and_return('target1')
       allow(self).to receive(:resource_group).and_return(group)
+      allow(self).to receive(:id).and_return('1')
```

Comment:
Could most of these stubs be replaced by a just a let and FactoryBot.create ?  Stubbing `:id` seems strange

Reply:
as I wrote, I do not see this code as something perfect, it is just a stub to make old specs work.